### ISSUE TITLE
keyError: 'next_results' on updated introduction__retweet_visualization.py

### ISSUE THREAD

**Issue by ez3kiel** [ptwobrussell/Mining-the-Social-Web#59]
Hello Matt, im having a problem with your new script introduction__retweet_visualization.py
Trying to run the script but i get an error:
next_results = search_results['search_metadata']['next_results']
KeyError: 'next_results'
Any idea? Thnx anyway

**Comment by ptwobrussell** [comment]
I think what could be happening is that what you are searching for is not returning any results, and that's causing an error. I should be handling that error case more elegantly, and will leave this ticket open as my note-to-self to do so.

Is it possible that this is what is happening? Can you share your query with me so I could try to reproduce and track down this problem?

**Comment by ez3kiel** [comment]
Yes, you're right!! my query is 'direngeziparkı' . i wonder why this was working with your previous script and no with the last one?

---

 Απο: Matthew A. Russell notifications@github.com
Προς: ptwobrussell/Mining-the-Social-Web Mining-the-Social-Web@noreply.github.com 
Κοιν.: ez3kiel giagoulas1711@yahoo.gr 
Στάλθηκε: 2:12 π.μ. Πέμπτη, 6 Ιουνίου 2013
Θέμα: Re: [Mining-the-Social-Web] keyError: 'next_results' on updated introduction__retweet_visualization.py (#59)

I think what could be happening is that what you are searching for is not returning any results, and that's causing an error. I should be handling that error case more elegantly, and will leave this ticket open as my note-to-self to do so.
Is it possible that this is what is happening? Can you share your query with me so I could try to reproduce and track down this problem?

**Comment by ptwobrussell** [comment]
Whether or not there are results for a query is dependent entirely upon Twitter. It may be the case that they are flushing a cache or only keeping a history of the most recent tweets for a particular time window. My understanding is that query results could be transient like that.

**Comment by wandeg** [comment]
I'm having the same problem with the script in Chapter 1

**Comment by ptwobrussell** [comment]
Thanks for letting me know. I will get this patched and updated in GitHub later this morning.

**Comment by ptwobrussell** [comment]
I believe this issue is fixed in commit 16a06591aa7c9283db9ca4158eb3c40960aaaa19 - can you test it out and close this issue if you agree? I believe this is the same issue as reported in #61
